### ISSUE TITLE
[Bug] Unable to create supported accessory (aqara temp)

### ISSUE THREAD

**Issue by martinj** [itavero/homebridge-z2m#19]
**Describe the bug**
Seems to be unable to create valid accessories. Acessories pop up in homekit but are marked as *Not supported*

Homebridge output
```
[10/18/2020, 5:04:29 PM] [zigbee2mqtt] Connecting to MQTT server at mqtt://localhost:1883
[10/18/2020, 5:04:29 PM] [zigbee2mqtt] Restoring accessory: temp/storage
[10/18/2020, 5:04:29 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0x00158d000540e656/get': ''
[10/18/2020, 5:04:29 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/bridge/config/devices/get': ''
```

This `[10/18/2020, 5:04:29 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0x00158d000540e656/get': ''` seems invalid do not now if thats related. But sending empty string to `device/get` generates error in Zigbee2Mqtt

From zigbee2mqtt output
```
Zigbee2MQTT:error 2020-10-18 16:04:29: Invalid JSON '', skipping...
````

**Expected behavior**

Expecting it to create a valid accessory (in this case a aqara temp sensor)

**MQTT messages**

* The device identification from `zigbee2mqtt/bridge/config/devices`

```javascript
{"dateCode":"20191205","description":"Aqara temperature, humidity and pressure sensor","friendly_name":"temp/storage","hardwareVersion":30,"ieeeAddr":"0x00158d000540e656","lastSeen":1603036668625,"manufacturerID":4151,"manufacturerName":"LUMI","model":"WSDCGQ11LM","modelID":"lumi.weather","networkAddress":11072,"powerSource":"Battery","softwareBuildID":"3000-0001","type":"EndDevice","vendor":"Xiaomi"}
```

**Versions used**
Please provide the version of the following pieces of software:
 - This plugin 0.0.7
 - Homebridge 1.2.3
 - Zigbee2mqtt 1.15.0
 - Homebridge Config UI X 4.29.0

**Comment by itavero** [comment]
Did the device publish its data/state while Homebridge was active?

I have these same sensors and for me they work without any issues.
They do *not* send updates very often (in my opinion), so you might try blowing into the sensor to temporarily increase the relative humidity (this usually triggers a status update).

I can have a look at the `get` query the plugin does, it used to work for me for most devices (not for battery powered ones I guess).
However, I think it is unrelated.

**Comment by itavero** [comment]
By the way, @martinj, I figured my question might be a bit strange if you don't know how this plugin works.
The plugin uses the keys and values from status messages in the plugin to deduct what kind of services a device offers.
This is briefly explained in the [_"How it (should 😉) work"_ section of the README](https://github.com/itavero/homebridge-z2m#how-it-should--work), beneath the mapping table.

**Comment by martinj** [comment]
Yes, I't did. I also tried with a motion sensor which send `occupancy` which i do believe you check for.

Tried once again, new install.
After plugin install, restarted homebridge.

Relevant output 
```
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Initializing zigbee2mqtt platform...
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Connecting to MQTT server at mqtt://localhost:1883
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/bridge/config/devices/get': ''
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Connected to MQTT server
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] New accessory: motion/front
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0x00124b001f9018ca/get': ''
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] New accessory: temp/storage
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0x00158d000540e656/get': ''
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] New accessory: 0xd0cf5efffe2a98de
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0xd0cf5efffe2a98de/get': ''
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] New accessory: contact/storage/door
[10/18/2020, 7:47:37 PM] [zigbee2mqtt] Publish to 'zigbee2mqtt/0x00124b001f90e856/get': ''
```

After startup, triggered motion, (don't mind the timestamps, some things are running in wrong timezones)

```
Zigbee2MQTT:info  2020-10-18 18:48:35: MQTT publish: topic 'zigbee2mqtt/motion/front', payload '{"battery":100,"battery_low":false,"linkquality":2,"occupancy":true,"tamper":false,"voltage":3200}'
Zigbee2MQTT:info  2020-10-18 18:49:35: MQTT publish: topic 'zigbee2mqtt/motion/front', payload '{"battery":100,"battery_low":false,"linkquality":13,"occupancy":false,"tamper":false,"voltage":3200}'
```

Devices config used
```
 "devices": [
                {
                    "id": "motion/front",
                    "exclude": false
                }
]
```

**Comment by itavero** [comment]
`occupancy` should indeed result in an Occupancy Sensor service being added.
Can you maybe enable debug logging (start homebridge with the `-D` flag)?
The plug-in should show which services are being created in the log.

I'm not sure if the slash in the name might somehow be causing this. I would be surprised if that's the case, but at first sight I don't see any noticable differences with my setup.

**Comment by martinj** [comment]
Tested with debug and when triggering motion I get: `Unhandled message on topic: motion/front`

The extra debug messages (everything else was the same)
```
[10/18/2020, 10:25:07 PM] [zigbee2mqtt] Unhandled message on topic: bridge/state
[10/18/2020, 10:25:07 PM] [zigbee2mqtt] Unhandled message on topic: bridge/config
[10/18/2020, 10:25:07 PM] [zigbee2mqtt] Unhandled message on topic: bridge/config/devices/get
[10/18/2020, 10:25:07 PM] [zigbee2mqtt] Skip Coordinator with IEEE address: 0x00124b0014d99b6d
[10/18/2020, 10:25:07 PM] [zigbee2mqtt] Unhandled message on topic: bridge/log

[10/18/2020, 10:26:38 PM] [zigbee2mqtt] Unhandled message on topic: motion/front
[10/18/2020, 10:27:38 PM] [zigbee2mqtt] Unhandled message on topic: motion/front
```

**Comment by martinj** [comment]
Ok it seems like my friendly names are the culprit, judging by this line https://github.com/itavero/homebridge-z2m/blob/a1386e99f8517b24bf5efb08ee91854fcba75c71/src/platform.ts#L110

Couldn't that be changed to 
```javascript
 } else if (this.accessories.some((acc) => acc.matchesIdentifier(topic))) {
   ...
}
```

**Comment by itavero** [comment]
I'll have a look at it later today. We can probably just remove the check all together and call the function that handles the update directly.
If there's no match it should still ignore the message.

**Comment by itavero** [comment]
I had another thought about it during my coffee break and I think I would ignore messages if:
* topic starts with `bridge/` (except for `bridge/config/devices` of course)
* topic ends with `/get` or `/set`

Otherwise, we can try to find a device that matches.

I'll probably have some time to implement and release that this week.

**Comment by itavero** [comment]
I've implemented the change I suggested on the `master` branch. I might be able to make a release today as well, but you should already be able to install it from the `master` branch to give it a try (see instructions in the README).

WIth regards to the `Invalid JSON` message in the zigbee2mqtt logs: apparently I'm also getting them.
I made an initial fix in the commit above by just putting an empty JSON object in the message, which at least gets rid off the message in the zigbee2mqtt logs.
I'll leave this issue open to improve this as well. If possible the `ServiceWrapper`s should be asked which keys should be requested.

**Comment by itavero** [comment]
@martinj I just published v0.0.8 which should solve your most important issue. 🙂

**Comment by martinj** [comment]
Tested and it works great now. Thanks! 👍
